### DISCUSSION TITLE
Curved mesh generation with second order elements

### DISCUSSION THREAD

**Discussion by lindsayad** [idaholab/moose#24484] [Q&A Meshing] [answered]
Do we have any mesh generators that create curved meshes (either 2D or 3D) with second order elements?

**Comment by oanaoana** [comment]
The TransfiniteMeshGenerator can do any curved edge, but I think is only Q1 for now. It can be changed to be Q2.

**Comment by GiudGiud** [comment]
Not a mesh generator but Cubit can get us that kind of meshes iirc

**Comment by roystgnr** [comment, marked as answer]
`FileMeshGenerator`, for curved inputs.  libMesh `build_sphere()` will do second-order interpolation on curved surfaces, and it looks like `SphereMeshGenerator` is calling that.  `MoveNodeGenerator` ought to get you curves if you're painstaking about it, or `ParsedNodeTransformGenerator` with more ease.

**Comment by aprilnovak** [comment]
We have some curved mesh generators in Cardinal, because Nek uses second order meshes. These are currently tailored to pin bundles by moving existing nodes onto a curved cylindrical surface. Despite the name, this is agnostic of anything special in Nek, we just use it often to make Nek meshes: https://cardinal.cels.anl.gov/source/meshgenerators/NekMeshGenerator.html 

Could potentially move that into the reactor module if it'd be of interest.

**Reply by lindsayad** [reply]
I think there definitely would be interest in getting more second order mesh capability as far upstream as possible. There's questions on Discussions about surface tension, surface curvature, etc. But I would't make this a personal priority of yours. You've got a ton of stuff on your plate haha
